test/NuGet.Clients.Tests/NuGet.VisualStudio.Common.Test/Telemetry/PackageSourceTelemetryTests.cs

```diff
@@ -21,81 +23,95 @@ public void Ctor_DuplicateSourceUrls_DoesNotThrow()
         {
             // Arrange
             const string feedUrl = "https://source.test/v3/index.json";
-            PackageSource[] sources = new[]
+            SourceRepository[] sources = new[]
             {
-                new PackageSource(source: feedUrl, name: "Source1"),
-                new PackageSource(source: feedUrl, name: "Source2")
+                new SourceRepository(new PackageSource(source: feedUrl, name: "Source1"), Repository.Provider.GetCoreV3()),
+                new SourceRepository(new PackageSource(source: feedUrl, name: "Source2"), Repository.Provider.GetCoreV3())
             };
 
             // Act
-            _ = new PackageSourceTelemetry(sources, Guid.Empty);
+            _ = new PackageSourceTelemetry(sources, Guid.Empty, "test");
 
             // Assert
             // no assert, just making sure ctor didn't throw.
         }
 
-        [Theory]
-        [InlineData("https://source.test/v3/flatcontainer/package/package.1.0.0.nupkg", true)]
-        [InlineData("https://source.test/v3/flatcontainer/package/index.json", false)]
-        public void AddAggregateData_SplitsMetadataAndNupkgCorrectly(string url, bool isNupkg)
+        [Fact]
+        public void AddResourceData_DifferentSources_StoredSeparately()
         {
             // Arrange
-            var pde = CreateSampleProtocolDiagnosticsEvent(url: new Uri(url));
+            var re1 = CreateSampleResourceEvent(source: "source1");
+            var re2 = CreateSampleResourceEvent(source: "source2");
             var data = new ConcurrentDictionary<string, PackageSourceTelemetry.Data>();
+            var stringTable = new ConcurrentDictionary<string, ConcurrentDictionary<string, string>>();
 
             // Act
-            PackageSourceTelemetry.AddAggregateData(pde, data);
+            PackageSourceTelemetry.AddResourceData(re1, data, stringTable);
```

Comment:
In this PR, `PackageSourceTelemetry.AddResourceData()` accepts 4 arguments. Here, you're using only 3.